server/src/main/java/org/elasticsearch/common/xcontent/AbstractObjectParser.java

```diff
@@ -220,13 +228,16 @@ public void declareRawObject(BiConsumer<Value, BytesReference> consumer, ParseFi
     private interface IOSupplier<T> {
         T get() throws IOException;
     }
+
     private static <T> List<T> parseArray(XContentParser parser, IOSupplier<T> supplier) throws IOException {
         List<T> list = new ArrayList<>();
-        if (parser.currentToken().isValue() || parser.currentToken() == XContentParser.Token.START_OBJECT) {
+        if (parser.currentToken().isValue() || parser.currentToken() == XContentParser.Token.VALUE_NULL ||
+            parser.currentToken() == XContentParser.Token.START_OBJECT) {
```

Comment:
Can you indent this one extra time? When the  body of the block below it and a part of the `if` statement line up I have trouble separating them properly when reading quickly.